### PR TITLE
Add warning about legacy glibc and bundled jvm

### PR DESCRIPTION
**Pull request by rjernst**
This commit adds a warning to the setup docs regarding using the bundled
jvm on centos 6 and oracle enterprise linux 6.

relates #62709

### CODE REVIEW

File: docs/reference/setup.asciidoc [hunk 1]
```diff
@@ -28,6 +28,10 @@ https://openjdk.java.net[OpenJDK] from the JDK maintainers (GPLv2+CE)
 within each distribution. The bundled JVM is the recommended JVM and
 is located within the `jdk` directory of the Elasticsearch home directory.
 
+WARNING: The bundled JVM requires glibc 2.14+. CentoOS 6 and Oracle Enterprise Linux 6
+have an outdated version of glibc, so you must use your own version of Java
```
Comment (mark-vieira) [inline docs/reference/setup.asciidoc:32]: Should we clarify that you must use a _compatible_ version of Java in those scenarios? Is there some JDK/Centos open issue we can link folks to for more background on this problem?

Comment (rjernst) [reply docs/reference/setup.asciidoc:32]: Compatible is a tricky word. Our compatibility matrix doesn't distinguish jdk support crossed by OS, so pointing them there won't help. And I don't know of any upstream issue we can reference. Would it help to clarify Java 14 or lower?

Comment (mark-vieira) [reply docs/reference/setup.asciidoc:32]: Yeah, I'd probably mention Java 14 or lower. That at least gives folks a better idea of what to do so they don't just go setting JAVA_HOME to a JDK 15 install and seeing the same problem.

**Comment by rjernst** [comment]
I added a mention of the required version. Can you take another look?

**Review by mark-vieira — APPROVED** [review]

**Pull request merged**